### ISSUE TITLE
Fix S1656 (`no-self-assign`): Add exception for `array.sort()`

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#1269]
https://jira.sonarsource.com/browse/RSPEC-1656
See https://community.sonarsource.com/t/javascript-false-positive-on-variable-self-assignment/6077

**Comment by ilia-kebets-sonarsource** [comment]
As [`array.sort()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#return_value) returns itself, modifying the array in place, the error message is correct, but might be misleading if you're not aware that it modifies the array in place.

**Comment by yassin-kammoun-sonarsource** [comment]
I am not a big fan of adding a custom message for `Array.prototype.sort()` because we would need to do the same for other mutating APIs. Besides, we already S4043 that covers this case. I would either close this ticket or add an exception to this rule for `Array.prototype.sort()` and `Array.prototype.reverse()`.

**Comment by ilia-kebets-sonarsource** [comment]
I like the idea of making an exception because it is already covered by S4043.

**Comment by ilia-kebets-sonarsource** [comment]
`Array.sort()` and `Array.reverse()` do not raise an issue for ESLint's `no-self-assign` anymore:

```js
let bar = [1,2,3];
bar = bar.sort((a, b) => a - b);
bar = bar.reverse();
```

Closing this
